app/models/plan.rb

```diff
@@ -61,10 +61,11 @@ class Plan < ActiveRecord::Base
 
   # Retrieves any plan organisationally or publicly visible for a given org id
   scope :organisationally_or_publicly_visible, -> (user) {
-    includes([:template, :roles])
+    includes([roles: :user])
       .where({
         visibility: [visibilities[:organisationally_visible], visibilities[:publicly_visible]],
-        "templates.org_id": user.org_id})
+        "roles.access": Role.access_values_for(:creator, :administrator, :editor, :commenter).min,
+        "users.org_id": user.org_id})
       .where(['NOT EXISTS (SELECT 1 FROM roles WHERE plan_id = plans.id AND user_id = ?)', user.id])
       .order(:title => :asc)
   }
```

Comment:
- Removing template from includes means that views/paginable/plans/_organisationally_or_publicly_visible.html.erb will introduce the (n+1) problem, i.e. for each plan it will be launched a query to retrieve the title of the template.
- Removing roles from includes does not have any effect since the view is using plan.owner which is still performing yet another query (we need to improve this)
- Replacing "templates.org_id: user.org_id in favour of "roles.access... and users.org_id" means that we want to retrieve any plan in the db whose access rights are "creator, administrator, editor and commenter" if the user of that plan belongs to the user's org. Do we really want that? I was thinking we want to retrieve only plans that belong to the user org and are marked "organisationally or publicly visible"

Reply:
- I will add template back to the `includes` list. The issue was specifically with the `template.org_id` in the where clause. That was causing it to only show plans that are derived from templates owned by the user's org. We want plans that were created by users belonging to the user's org regardless of what template was used to create the plan. For example If we both belong to UoE and I create a Horizon 2020 (and make it organisationally visible) you should see it when you log in. Right now, no one would see it because the template belongs to Horizon2020
- The code is already running the plan.owner query before this proposed change. I'm not attempting to refactor the entire process, just address the one bug
- Yes, that is what we want to show. Plans have no direct association to Orgs so we have to go through the Roles table.